lms/djangoapps/grades/models.py

```diff
@@ -302,6 +302,13 @@ def _is_unattempted_with_score(self):
         """
         return self.first_attempted is None and any(field != 0.0 for field in (self.earned_all, self.earned_graded))
 
+    def full_clean(self, exclude=None, validate_unique=True):
+        # efischer debug, nomerge
+        from nose.tools import set_trace; set_trace()
+        self.clean_fields(exclude='visible_blocks')
```

Comment:
`clean_fields` is hitting the database to ensure the `visible_blocks` hash exists (since it's a FK), that's the source of 1 of the 2 extra queries per subsection.

```
> /edx/app/edxapp/edx-platform/lms/djangoapps/grades/models.py(308)full_clean()
-> self.clean_fields()
(Pdb) n
DEBUG:django.db.backends:(0.000) SELECT (1) AS `a` FROM `grades_visibleblocks` WHERE `grades_visibleblocks`.`hashed` = 'UTKduqVDW7NxFiqu197P9/Yd0Bw=' LIMIT 1; args=(u'UTKduqVDW7NxFiqu197P9/Yd0Bw=',)
```